editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/util/JobWaiterUtil.java

```diff
@@ -0,0 +1,70 @@
+/******************************************************************************* 
+ * Copyright (c) 2016 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/ 
+package org.fusesource.ide.projecttemplates.util;
+
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.core.runtime.jobs.Job;
+import org.fusesource.ide.projecttemplates.internal.ProjectTemplatesActivator;
+
+public class JobWaiterUtil {
+	
+	boolean isEndless = false;
+	
+	private Throwable exception = null;
+
+	public void waitBuildAndRefreshJob(IProgressMonitor monitor) {
+		waitBuildAndRefreshJob(20, monitor);
+	}
+	
+	private void waitBuildAndRefreshJob(int decreasingCounter, IProgressMonitor monitor) {
+		int currentCounter;
+		if(isEndless){
+			currentCounter = decreasingCounter;
+			ProjectTemplatesActivator.pluginLog().logInfo("log trace to ensure it is not looping");
+		} else {
+			if (decreasingCounter <= 0) {
+				if(exception != null){
+					ProjectTemplatesActivator.pluginLog().logWarning(exception);
```

Comment:
I wasn't sure as we are trying to workaround an Eclipse bugs but as you think it is better and I wasn't sure, seems best to log as error :-)